fed/mosaic-sumo/src/main/java/org/eclipse/mosaic/fed/sumo/util/SumoRouteFileCreator.java

```diff
@@ -64,138 +59,138 @@
-            routeFileDocument.appendChild(parent);
+            Node parent = document.createElement("routes");
+            document.appendChild(parent);
+            return document;
         } catch (ParserConfigurationException e) {
             log.warn("Couldn't instantiate DocumentBuilder, this will result in no Route-File being written");
         }
+        return null;
     }
 
     /**
-     * Stores the existing types and routes from the given file before deletion.
+     * Adds the given vehicle types including the already existing
+     * to the {@link #departureDocument}.
      */
-    private void extractVehicleTypesAndRoutes(File baseRouteFile) {
-        log.info("Reading routes and vTypes before Rewriting.");
-        vehicleTypesWithAttributes = readVehicleTypes(baseRouteFile);
-        routes = readRoutes(baseRouteFile);
-    }
+    @SuppressWarnings("UnusedReturnValue")
+    public SumoRouteFileCreator addVehicleTypes(Map<String, VehicleType> additionalVTypes) {
+        // adding new types with additional parameters from sumo config
+        Map<String, Map<String, String>> newVTypes = generateAttributesMap(additionalVTypes, timeGapOffset);
 
-    private Map<String, Map<String, String>> readVehicleTypes(File baseRouteFile) {
-        VehicleTypeParser vehTypeParser = new VehicleTypeParser();
+        // applies the additional para
+        applyParametersFromSumoConfiguration(newVTypes);
 
-        if (baseRouteFile != null && baseRouteFile.exists()) {
-            try (InputStream input = new FileInputStream(baseRouteFile)) {
-                SAXParserFactory.newInstance().newSAXParser().parse(input, vehTypeParser);
-            } catch (Exception e) {
-                log.error("Could not read vehicle type definition", e);
-            }
-        }
-        return vehTypeParser.vehTypeAttributes;
+        // write vehicle type to new route file containing vehicle types
+        writeVehicleTypes(newVTypes);
+
+        return this;
     }
 
-    /**
-     * This method reads vehicle routes from the rou-file.
-     *
-     * @return routes with their IDs.
-     */
-    private Map<String, VehicleRoute> readRoutes(File baseRouteFile) {
-        RouteParser routeParser = new RouteParser();
-
-        if (baseRouteFile != null && baseRouteFile.exists()) {
-            try (InputStream input = new FileInputStream(baseRouteFile)) {
-                SAXParserFactory.newInstance().newSAXParser().parse(input, routeParser);
-            } catch (Exception e) {
-                log.error("Could not read vehicle type definition", e);
+    private void applyParametersFromSumoConfiguration(Map<String, Map<String, String>> newVTypes) {
+        for (Entry<String, Map<String, String>> sumoParameterEntry : additionalVTypeParameters.entrySet()) {
+            String currentVehicleType = sumoParameterEntry.getKey();
+            newVTypes.putIfAbsent(currentVehicleType, new HashMap<>());
```

Comment:
It is also possible to define new vTypes in the `sumo_config.json`. This is because vTypes defined in Mapping are only forwarded to SUMO if `VehicleSpawner`s of that vType are defined. Additionally this allows overwriting the `DEFAULT_VEHTYPE`.